Examples/MAX78000/EEPROM_Emulator/src/cache.c

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+* Copyright (C) 2022 Maxim Integrated Products, Inc., All Rights Reserved.
+*
+* Permission is hereby granted, free of charge, to any person obtaining a
+* copy of this software and associated documentation files (the "Software"),
+* to deal in the Software without restriction, including without limitation
+* the rights to use, copy, modify, merge, publish, distribute, sublicense,
+* and/or sell copies of the Software, and to permit persons to whom the
+* Software is furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included
+* in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+* OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+* OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+* ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+* OTHER DEALINGS IN THE SOFTWARE.
+*
+* Except as contained in this notice, the name of Maxim Integrated
+* Products, Inc. shall not be used except as stated in the Maxim Integrated
+* Products, Inc. Branding Policy.
+*
+* The mere transfer of this software does not imply any licenses
+* of trade secrets, proprietary technology, copyrights, patents,
+* trademarks, maskwork rights, or any other form of intellectual
+* property whatsoever. Maxim Integrated Products, Inc. retains all
+* ownership rights.
+*******************************************************************************
+*/
+
+#include "cache.h"
+
+#include <stdio.h>
+#include <string.h>
+#include "flc.h"
+
+int cache_init(cache_t *cache, uint32_t init_addr)
+{
+    int err;
+
+    if (cache == NULL) {
+        return E_NULL_PTR;
+    } else if (init_addr < MXC_FLASH_MEM_BASE ||
+               init_addr > (MXC_FLASH_MEM_BASE + MXC_FLASH_MEM_SIZE)) {
+        return E_BAD_PARAM;
+    }
+
+    // Configure FLC
+    err = MXC_FLC_Init();
+    if (err != E_NO_ERROR) {
+        printf("Failed to initialize flash controller.\n");
+        return err;
+    }
+
+    // Get starting address of flash page
+    init_addr -= init_addr % MXC_FLASH_PAGE_SIZE;
+
+    // Initialize cache values and starting address
+    memcpy((void *)cache->cache, (void *)init_addr, MXC_FLASH_PAGE_SIZE);
```

Comment:
Shouldn't need to cast to `(void *)` right?  `void *` as an argument type means a pointer to anything should be accepted here without needing to cast it

Reply:
init_addr needs to be casted since it's not a pointer. You're right about the other one though, so I removed it.